web-admin/src/routes/[organization]/[project]/-/settings/public-urls/+page.svelte

```diff
@@ -0,0 +1,102 @@
+<script lang="ts">
+  import PublicURLsTable from "@rilldata/web-admin/features/public-urls/PublicURLsTable.svelte";
+  import { page } from "$app/stores";
+  import {
+    createAdminServiceListMagicAuthTokens,
+    getAdminServiceListMagicAuthTokensQueryKey,
+    createAdminServiceRevokeMagicAuthToken,
+  } from "@rilldata/web-admin/client";
+  import NoPublicURLCTA from "@rilldata/web-admin/features/public-urls/NoPublicURLCTA.svelte";
+  import { useQueryClient } from "@tanstack/svelte-query";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import DelayedSpinner from "@rilldata/web-common/features/entity-management/DelayedSpinner.svelte";
+  import { useDashboardsV2 } from "@rilldata/web-admin/features/dashboards/listing/selectors";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+
+  $: organization = $page.params.organization;
+  $: project = $page.params.project;
+
+  let pageSize = 16;
+  let pageToken: string | undefined = undefined;
+
+  $: magicAuthTokens = createAdminServiceListMagicAuthTokens(
+    organization,
+    project,
+    {
+      pageSize,
+      pageToken,
+    },
+  );
+
+  $: dashboards = useDashboardsV2($runtime.instanceId);
+
+  // Reset and update allTokensStore when magicAuthTokens changes
+  $: currentPageTokens =
+    $magicAuthTokens.data && $dashboards.data
+      ? $magicAuthTokens.data.tokens.map((token) => {
+          const dashboard = $dashboards.data.find(
+            (d) => d.resource.meta.name.name === token.metricsView,
+          );
+          return {
+            ...token,
+            dashboardTitle:
+              dashboard?.resource.metricsView.spec.title || "Unknown Dashboard",
```

Comment:
1. We don't currently do it throughout the codebase, but FYSA that we _should_ reference the Metrics View's `state.validSpec` rather than it's `spec`
2. When there's no `title` we can fallback to the `name`, which is required: `dashboard?.resource.metricsView.spec.title || dashboard?.resource.meta.name.name`

Reply:
Okay, will use validSpec's title

```
dashboardTitle: dashboard?.resource.metricsView.state?.validSpec?.title || dashboard?.resource.meta.name.name,
```